src/main/java/com/devonfw/ide/sonarqube/common/api/config/DevonArchitecturePackage.java

```diff
@@ -1,129 +1,158 @@
 package com.devonfw.ide.sonarqube.common.api.config;
 
-import com.devonfw.module.basic.common.api.reflect.Devon4jPackage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class DevonArchitecturePackage {
 
-  private Devon4jPackage devon4jPackage;
+  private Packages packages;
 
-  public DevonArchitecturePackage(String packageName) {
+  private Pattern pattern;
 
-    this.devon4jPackage = Devon4jPackage.of(packageName);
-  }
+  private Pattern layerPattern;
 
-  public int getSegmentCount() {
+  private Pattern scopePattern;
 
-    return this.devon4jPackage.getSegmentCount();
-  }
+  private Matcher matcher;
 
-  public String getSegment(int index) {
+  private static final Logger logger = Logger.getLogger("logger");
 
-    return this.devon4jPackage.getSegment(index);
-  }
+  /**
+   *
+   * The constructor.
+   *
+   * @param packageName Name of the package to check for validity
+   * @param packages Packages object containing the info provided by the user inside architecture.json
+   */
+  public DevonArchitecturePackage(String packageName, Packages packages) {
 
-  public boolean isValid() {
+    this.packages = packages;
+    this.packages.setPattern(replaceMappings(packages.getPattern()));
```

Comment:
we should not manipulate the `Packages` object here. See comment on `replaceMappings`.